File: Project/Utils/DynamicDirections.java
package Utils;

import java.util.HashMap;
import java.util.Map;

public class DynamicDirections {
    private static final Map<String, DynamicDirections> instances = new HashMap<>();
    private static int nextOrdinal = 0;
    private final String name;
    private final int ordinal;

    private DynamicDirections(String name) {
        this.name = name;
        this.ordinal = nextOrdinal++;
    }

    public static DynamicDirections addInstance(String name) {
        DynamicDirections instance = new DynamicDirections(name);
        instances.put(name, instance);
        return instance;
    }

    public static DynamicDirections get(String name) {
        return instances.get(name);
    }

    public int ordinal() {
        return ordinal;
    }

    @Override
    public String toString() {
        return name;
    }

    public static DynamicDirections[] values() {
        return instances.values().toArray(new DynamicDirections[0]);
    }

    public static void clear() {
        instances.clear();
        nextOrdinal = 0;
    }
}

File: Project/Utils/DynamicDirections.py
class DynamicDirections:
    __instances = {}
    __next_ordinal = 0

    def __init__(self, name):
        self.__name = name
        self.__ordinal_value = DynamicDirections.__next_ordinal
        DynamicDirections.__next_ordinal += 1

    @classmethod
    def addInstance(cls, name):
        instance = DynamicDirections(name)
        cls.__instances[name] = instance
        return instance

    @classmethod
    def get(cls, name):
        return cls.__instances.get(name)

    def ordinal(self):
        return self.__ordinal_value

    def __str__(self):
        return self.__name

    @classmethod
    def values(cls):
        return list(cls.__instances.values())

    @classmethod
    def clear(cls):
        cls.__instances.clear()
        cls.__next_ordinal = 0
